### PR TITLE
couchdb setup

### PR DESCRIPTION
**Pull request by maxlath**
setting up the basic tools to work with couchdb

### CODE REVIEW

File: server/lib/db.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const __ = CONFIG.universalPath
+const _ = __.require('lib', 'utils')
+const dbC = CONFIG.db
+const fullUrl = `${dbC.protocol}://${dbC.username}:${dbC.password}@${dbC.host}:${dbC.port}`
+const nano = require('nano-blue')(fullUrl)
+
+module.exports = function (dbName) {
+  const db = nano.use(dbName)
+
+  return db.info()
```
Comment (almereyda) [inline server/lib/db.js]: explain higher order functions here

**Comment by almereyda** [comment]
@maxlath You may also want to extract the `universal_path` module into a separate npm package due to sharing code with Inventaire.

File: server/lib/db.js (hunk 1 above)
Comment (maxlath) [inline server/lib/db.js]: @almereyda an Higher order function is just a function that returns a function.
For instance:

``` javascript
const Hello = function (name) { console.log(`hello ${name}!`) }
const hello = Hello('Jon')
hello()
// => hello Jon!
```

Here, `Hello` is the higher order function. Usually, capitals are reserved to classes, but as I'm using very few of those, I tend to use them for Higher order functions.
Cf the [`inv-loggers`](http://github.com/inventaire/inv-loggers) module that gives access to `Log` and `log` functions, with the latter being just a wrapper for the first

Resources:
- [Higher-order functions - Part 1 of Functional Programming in JavaScript](https://www.youtube.com/watch?v=BMUiFMZr7vk&index=1&list=PL0zVEGEvSaeEd9hlmCXrk5yUyqUag-n84)
- all [Crockford on JavaScript](https://www.youtube.com/watch?v=JxAXlJEmNMg&list=PL7664379246A246CB)

**Comment by species** [comment]
Pair-reviewed and OK.

**Pull request merged**